pkg/util/kubernetes/apiserver/hpa_controller_test.go

```diff
@@ -314,25 +454,25 @@ func TestAutoscalerSync(t *testing.T) {
 func TestAutoscalerControllerGC(t *testing.T) {
 	testCases := []struct {
 		caseName string
-		metrics  []custommetrics.ExternalMetricValue
+		metrics  map[string]custommetrics.ExternalMetricValue
 		hpa      *v2beta1.HorizontalPodAutoscaler
 		expected []custommetrics.ExternalMetricValue
 	}{
 		{
 			caseName: "hpa exists for metric",
-			metrics: []custommetrics.ExternalMetricValue{
-				{
+			metrics: map[string]custommetrics.ExternalMetricValue{
+				"external_metric-default-fobo-requests_per_s": {
 					MetricName: "requests_per_s",
 					Labels:     map[string]string{"bar": "baz"},
-					HPA:        custommetrics.ObjectReference{Name: "foo", Namespace: "default", UID: "1111"},
+					HPA:        custommetrics.ObjectReference{Name: "fobo", Namespace: "default", UID: "1111"},
```

Comment:
any reason behind this change? 🤔

Reply:
No good reason. I made one typo and decided to update the whole unit test with the typo. Changing back